uaclient/clouds/aws.py

```diff
@@ -43,10 +42,33 @@ def _get_imds_url_response(self):
     @property  # type: ignore
     @util.retry(exceptions.CloudMetadataError, retry_sleeps=[0.5, 1, 1])
     def identity_doc(self) -> Dict[str, Any]:
-        imds_url_response = self._get_imds_url_response()
+        headers = self._request_imds_v2_token_headers()
+        url = IMDS_URL.format(self._ip_address)
+        imds_url_response = self._get_imds_url_response(url, headers=headers)
         secret_manager.secrets.add_secret(imds_url_response)
         return {"pkcs7": imds_url_response}
 
+    @util.retry(exceptions.CloudMetadataError, retry_sleeps=[0.5, 1, 1])
+    def _get_ii_doc(self) -> Dict:
+        headers = self._request_imds_v2_token_headers()
+        url = _IMDS_IID_URL.format(self._ip_address)
+        try:
+            ii_doc = json.loads(
+                self._get_imds_url_response(url, headers=headers)
+            )
+        except json.JSONDecodeError as e:
+            LOG.debug("Error decoding instance identity document: %s", e)
+            return {}
+        return ii_doc
+
+    @property  # type: ignore
+    def is_likely_pro(self) -> bool:
```

Comment:
:rofl: good function name haha

Reply:
Do you have any better suggestion? I think it represents right the feature it holds.

Reply:
no this was an actual compliment, keep it